libs/core/src/components/pds-select/pds-select.tsx

```diff
@@ -0,0 +1,146 @@
+
+  /**
+   * Indicates whether or not the select field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * The value of the select field.
+   */
+  @Prop({ mutable: true }) value?: string;
+
+  /**
+   * Emitted when a keyboard input occurred.
+   */
+  @Event() pdsSelect: EventEmitter<InputEvent>;
+
+  /**
+   * Handles the select event for the component.
+   *
+   * @param ev - The event object triggered by the select action.
+   * @remarks
+   * This method casts the event target to an HTMLSelectElement and updates the component's value
+   * with the selected option's value. It then emits a custom event (`pdsSelect`) with the original event.
+   */
+  private onSelectEvent = (ev: Event) => {
+    const select = ev.target as HTMLSelectElement;
+    this.value = select.value;
+    this.pdsSelect.emit(ev as InputEvent);
+  };
+
+  /**
+   * Handles the change event for the slot element.
+   * This method is triggered when the slot content changes.
+   * It updates the inner HTML of the select element by cloning and appending
+   * the assigned <option> elements from the slot.
+   */
+  private handleSlotChange = () => {
+    const slot = this.slotContainer.querySelector('slot') as HTMLSlotElement;
+
+    this.selectEl.innerHTML = '';
+    const assignedElements = slot.assignedElements({ flatten: true }) as HTMLOptionElement[];
+
+    assignedElements.map((item: HTMLOptionElement) => {
+      const option = item;
+
+      if (option.tagName === 'OPTION') {
+        this.selectEl.appendChild(option.cloneNode(true));
+      }
+    });
+  };
+
+  render() {
+    return (
+      <Host aria-disabled={this.disabled ? 'true' : null}>
+        <div class="pds-select">
+          <PdsLabel htmlFor={this.componentId} text={this.label} />
+          <select
+            autocomplete={this.autocomplete}
+            class="pds-select__field"
+            disabled={this.disabled}
+            id={this.componentId}
+            name={this.name}
+            onChange={this.onSelectEvent}
+            required={this.required}
+            ref={(el) => (this.selectEl = el as HTMLSelectElement)}
+          ></select>
+          <div aria-hidden="true" class="hidden" ref={(el) => (this.slotContainer = el)}>
+            <slot onSlotchange={this.handleSlotChange}></slot>
+          </div>
+          {(this.helperMessage || this.errorMessage) && (
+            <div class="pds-select__message">
+              {this.helperMessage && (
+                <p class="pds-select__helper-message" id={messageId(this.componentId, 'helper')}>
+                  {this.helperMessage}
+                </p>
+              )}
+              {this.errorMessage && (
+                <p class="pds-select__error-message" id={messageId(this.componentId, 'error')} aria-live="assertive">
+                  <pds-icon icon={danger} size="small" />
+                  {this.errorMessage}
+                </p>
+              )}
+            </div>
+          )}
```

Comment:
Is it intended to have both a helper message and error message display when both are set?

Reply:
Good question, Ashley! I would imagine we would want both helper and error messages for better context, but I would be open to changing if that's what we all deem the best option.

Reply:
Here's the relevant Slack thread for this, in case we want an action item: https://kajabi.slack.com/archives/C03GKQJHCLX/p1732319349764399